CompilerSource/compiler/compile.cpp

```diff
@@ -194,11 +194,14 @@ static bool ends_with(std::string const &fullString, std::string const &ending)
 }
 
 int lang_CPP::compile(const GameData &game, const char* exe_filename, int mode) {
-  std::string exename = std::string(exe_filename);
-  const std::string buildext = compilerInfo.exe_vars["BUILD-EXTENSION"];
-  if (!ends_with(exename, buildext)) {
-    exename += buildext;
-    exe_filename = exename.c_str();
+  std::string exename = "";
```

Comment:
This is declared outside the braces, because we use it as a buffer to assign the contents of exe_filename inside the block. Therefore, it needs to live just as long.